updater/reports/ReportOrgsAbandoned.py

```diff
@@ -0,0 +1,38 @@
+from .ReportDaily import *
+
+# Find the organizations that have not received a push for the longest time.
+# Only look at organizations that have not received a push for at least one
+# year. Only look at repositories that are still maintained (not archived!).
+class ReportOrgsAbandoned(ReportDaily):
+	def name(self):
+		return "organizations-abandoned"
+
+	def updateDailyData(self):
+		self.detailedHeader, self.detailedData = self.parseData(self.executeQuery(self.query()))
+		if len(self.data) == 0:
+			self.header = ["date", "abandoned organizations"]
+		self.data.append([str(self.yesterday()), len(self.detailedData)])
+		self.truncateData(self.timeRangeTotal())
+		self.sortDataByDate()
+
+	def query(self):
+		query = '''
+			SELECT
+				users.login AS "organization",
+				date(MAX(pushes.created_at)) AS "last push"
+			FROM
+				repositories
+				JOIN users ON repositories.owner_id = users.id
+				JOIN pushes ON pushes.repository_id = repositories.id
+			WHERE
+				    users.type = "organization"
+				AND repositories.maintained = 1 ''' + \
+				self.andExcludedEntities("users.login") + '''
+			GROUP BY
+				users.id
+			HAVING
+				MAX(pushes.created_at) < DATE_SUB(NOW(), INTERVAL 1 YEAR)
```

Comment:
I’d prefer using the Python-based approach to selecting time ranges instead of using `DATE_SUB(NOW())` and `INTERVAL 1 YEAR`.

The main reason for this is that I think it would be better to exclude today’s data from the report and to start with yesterday instead.